docs/adrs/v3-setup-dotnet.md

```diff
@@ -0,0 +1,48 @@
+  - Ubuntu:`/home/runner/.dotnet`
+  - Windows: `C:\Users\runneradmin\AppData\Local\Microsoft\dotnet`
+- It means that action always download and install .NET version even if it is pre-installed. Also after using the action all pre-installed .NET versions are unavailable because `DOTNET_ROOT` is overriden to user directory.
+The behavior is different for macOS runners because pre-installation directory matches the one that is used by action. It means action can use pre-installed versions if it exists, which speeds up customer's workflow.
+
+- The different behavior of setup task on Ubuntu, Windows and MacOS runners is unclear and confusing for customers.
+
+- .NET supports installing and using multiple versions of .NET SDK and .NET runtime sidy-by-side. .NET CLI will use the latest of installed .NET SDK and .NET runtime version if there is no global.json file containing a different version. This behavior is defined by .NET design (https://docs.microsoft.com/en-us/dotnet/core/versions/selection).
+
+- The action contains logic to handle inputs with wildcards, for example `5.0.x`, `5.0.*`, `5.x` or `5.*`. This logic uses metadata from `https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json` to retrieve the list of available releases and get the latest release version for the specified major and/or minor version from input. After that, installer script (`dotnet-install.ps1` for Windows or `dotnet-install.sh` for Linux and MacOS) installs required SDK using exact version as a parameter.
+
+# Proposal
+- Change .NET installation path for Windows and Ubuntu images to match the location of pre-installed versions by using `-InstallDir` (Windows) and `--install-dir` (Ubuntu) property for installer scripts:
+https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script
+- Get rid of redundant logic to resolve wildcard version on Action side and start relying on [official installer scripts provided by .NET Core team](https://github.com/dotnet/install-scripts).  
+The execution `dotnet-install.ps1 -Channel 5.0` installs the latest patch version for 5.0 sdk. In this way we can handle inputs with wildcard as patch version (`5.0.x` or `5.0.*`) by passing major and minor version to installer script directly as `channel` parameter. This parameter supports two-part version in X.Y format as input values ([see installer scripts documentation](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script)). In further the version input format of action will follow the format which is supported by installers scripts. 
+
+# Breaking changes
+- The presence of pre-installed .NET versions that are higher than version that the users specify in the setup task can be breaking for some customers, who expect only one installed .NET version on runner after using setup task. If user doesn't have .NET version specified in project file, `dotnet` will use the latest installed version instead of provided in setup task.  
```

Comment:
The presence of pre-installed .NET versions that are higher than the versions that users specify in the setup task can be breaking for some customers, who expect only one installed .NET version on the runner after using the setup task. If the user doesn't have .NET version specified in the project file, `dotnet` will use the latest installed version instead of the one provided in the setup task.